browser_patches/firefox-beta/juggler/NetworkObserver.js

```diff
@@ -533,6 +533,9 @@ class NetworkRequest {
       // remoteAddress is not defined for cached requests.
     }
 
+    const fromServiceWorker = this._networkObserver._channelIdsFulfilledByServiceWorker.has(this.requestId);
+    this._networkObserver._channelIdsFulfilledByServiceWorker.delete(this.requestId);
+
```

Comment:
Do we need to remove request id from the set on failure as well?

Reply:
Since we listen to "service-worker-synthesized-response", I think it only happens when response is created, not on complete failures.